pkg/solana/client/multinode/transaction_sender.go

```diff
@@ -154,80 +155,81 @@ func (txSender *TransactionSender[TX, CHAIN_ID, RPC]) SendTransaction(ctx contex
 	return txSender.collectTxResults(ctx, tx, healthyNodesNum, txResults)
 }
 
-func (txSender *TransactionSender[TX, CHAIN_ID, RPC]) broadcastTxAsync(ctx context.Context, rpc RPC, tx TX) sendTxResult {
-	txErr := rpc.SendTransaction(ctx, tx)
-	txSender.lggr.Debugw("Node sent transaction", "tx", tx, "err", txErr)
-	resultCode := txSender.txErrorClassifier(tx, txErr)
+func (txSender *TransactionSender[TX, RESULT, CHAIN_ID, RPC]) broadcastTxAsync(ctx context.Context, rpc RPC, tx TX) RESULT {
+	result := rpc.SendTransaction(ctx, tx)
+	txSender.lggr.Debugw("Node sent transaction", "tx", tx, "err", result.TxError())
+	resultCode := txSender.txErrorClassifier(tx, result.TxError())
 	if !slices.Contains(sendTxSuccessfulCodes, resultCode) {
-		txSender.lggr.Warnw("RPC returned error", "tx", tx, "err", txErr)
+		txSender.lggr.Warnw("RPC returned error", "tx", tx, "err", result.TxError())
 	}
-	return sendTxResult{Err: txErr, ResultCode: resultCode}
+	result.SetCode(resultCode)
```

Comment:
There is no need for us to `SetCode` here. It can be done by the RPC inside `SendTransaction` impl